sources/settings-extensions/network/src/generate.rs

```diff
@@ -0,0 +1,54 @@
+/// Generators for network settings
+use error::{ExecutionSnafu, GenerateError, InvalidHostnameSnafu};
+use modeled_types::ValidLinuxHostname;
+use snafu::ResultExt;
+use std::process::Command;
+
+pub fn generate_hostname() -> std::result::Result<ValidLinuxHostname, GenerateError> {
+    let ret = Command::new("netdog")
+        .arg("generate-hostname")
+        .output()
+        .context(ExecutionSnafu {})?;
+
+    if !ret.status.success() {
+        Err(GenerateError::Failure {
+            message: String::from_utf8_lossy(&ret.stderr).to_string(),
+        })
+    } else {
+        let hostname = parse_stdout(String::from_utf8_lossy(&ret.stdout).to_string());
+        Ok(ValidLinuxHostname::try_from(hostname).context(InvalidHostnameSnafu {})?)
```

Comment:
Can the braces be eliminated in these zero params Snafu struct constructions? (can't remember)

```suggestion
        Ok(ValidLinuxHostname::try_from(hostname).context(InvalidHostnameSnafu)?)
```

Reply:
looks like they can! updating